sql/src/main/java/io/crate/analyze/OptimizeTableAnalyzer.java

```diff
@@ -90,4 +92,22 @@ public OptimizeTableAnalyzedStatement analyze(OptimizeStatement stmt, Analysis a
         }
         return indexNames;
     }
+
+    private void validateSettings(Settings settings) {
+        if (settings.getAsBoolean(UPGRADE_SEGMENTS.name(), UPGRADE_SEGMENTS.defaultValue())) {
+            if (!settings.getAsBoolean(FLUSH.name(), FLUSH.defaultValue())) {
```

Comment:
is it really needed/worth doing the concrete checks? wouldn't it be enough to check that `upgrade_segments` is the only setting (by checking the settings size additionally)?